### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

File: src/modelmanager.cpp [hunk 1]
```diff
@@ -1343,7 +1343,7 @@ Status ModelManager::readAvailableVersions(std::shared_ptr<FileSystem>& fs, cons
 Status ModelManager::addModelVersions(std::shared_ptr<ovms::Model>& model, std::shared_ptr<FileSystem>& fs, ModelConfig& config, std::shared_ptr<model_versions_t>& versionsToStart, std::shared_ptr<model_versions_t> versionsFailed) {
     Status status = StatusCode::OK;
     try {
-        status = model->addVersions(versionsToStart, config, fs, *ieCore, versionsFailed, this->metricRegistry, &this->metricConfig);
```
Comment (atobiszei) [inline src/modelmanager.cpp:1346]: Replace std::shared_ptr<model_versions_t> versionsFailed)  with
std::shared_ptr<model_versions_t>& versionsFailed, or even better just with
model_versions_t& versionsFailed

Comment (atobiszei) [reply src/modelmanager.cpp:1346]: The same is true for reloadModelVersions

File: src/server.cpp [hunk 2]
```diff
@@ -276,6 +276,11 @@ Status Server::startModules(ovms::Config& config) {
         it = modules.find(PYTHON_INTERPRETER_MODULE_NAME);
         if (it != modules.end()) {
             auto pythonModule = dynamic_cast<const PythonInterpreterModule*>(it->second.get());
+            if (!pythonModule) {
+                SPDLOG_ERROR("PythonInterpreterModule not initialized");
+                return StatusCode::INTERNAL_ERROR;
+            }
+
```
[the remark this answers is not in the document]
Comment (atobiszei) [reply src/server.cpp:283]: I vote for option 3

File: src/test/c_api_tests.cpp [hunk 3]
```diff
@@ -20,6 +20,7 @@
 #include <memory>
 #include <string>
 #include <thread>
+#include <utility>
```
Comment (atobiszei) [inline src/test/c_api_tests.cpp:23]: Why do we need it here?

**Review by atobiszei — APPROVED** [review]

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**